src/solc_0.8/common/Base/TheSandbox712.sol

```diff
@@ -3,13 +3,14 @@ pragma solidity 0.8.2;
 
 contract TheSandbox712 {
     bytes32 internal constant EIP712DOMAIN_TYPEHASH =
-        keccak256("EIP712Domain(string name,string version,address verifyingContract)");
+        keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
     // solhint-disable-next-line var-name-mixedcase
     bytes32 public immutable _DOMAIN_SEPARATOR;
 
     constructor() {
         _DOMAIN_SEPARATOR = keccak256(
-            abi.encode(EIP712DOMAIN_TYPEHASH, keccak256("The Sandbox"), keccak256("1"), address(this))
+            // chainId 137 = Polygon
+            abi.encode(EIP712DOMAIN_TYPEHASH, keccak256("The Sandbox"), keccak256("1"), 137, address(this))
```

Comment:
137 should be sent through the constructor or with block.chainId ?

https://docs.soliditylang.org/en/v0.8.0/units-and-global-variables.html#block-and-transaction-properties

Reply:
block.chainId works fine. 

Done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/641/commits/b687d530f6a6edf2ab94c75c9efcae6a0e4242df